### PR TITLE
Fix(eos_config_deploy_cvp): avoid making CV_DEVICES becoming None

### PR DESCRIPTION
**Pull request by noredistribution**
… devices

## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_config_deploy_cvp`

## Proposed changes
in https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2
if the device_filter does not match any devices inside the for loop `CVP_DEVICES` will be empty, so it'll become `None` and then on the ansible-cvp side we'll get that TypeError traceback mentioned in the #1293 

This will only set `CVP_DEVICES` to an empty list, and then validation should be done in the cv_devices_v3 module on ansible-cvp side

## How to test
To check if the list will become empty simply  print the `CVP_DEVICES` var in https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_config_deploy_cvp/tasks/v3/present.yml after L41
like this:
```
- name: "Print CVP_DEVICES_RESULTS"
  ansible.builtin.debug:
    var: CVP_DEVICES
```

which should result in

```
TASK [arista.avd.eos_config_deploy_cvp : Configure devices on AVD-CVP-1] **********************************************************
ok: [AVD-CVP-1]

TASK [arista.avd.eos_config_deploy_cvp : Print CVP_DEVICES_RESULTS] ***************************************************************
ok: [AVD-CVP-1] => {
    "CVP_DEVICES": []
}
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2 [hunk 1]
```diff
@@ -39,5 +44,5 @@ CVP_CONTAINERS:
 {%         for configlet in hostvars[groups[container_root][0]]['cv_configlets']['containers'][container_name] | arista.avd.natural_sort %}
       - {{ configlet }}
 {%         endfor %}
-{%      endif %}
+{%     endif %}
 {% endfor %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2]: New line required at the end of file.

Comment (noredistribution) [reply ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2]: done thx!

File: ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2 [hunk 2]
```diff
@@ -5,15 +5,17 @@
 {% else %}
 {%    set local_var.device_filter = device_filter %}
 {% endif %}
+{% set dev_cntr = namespace(value=0) %}
 ---
 CVP_DEVICES:
 {% for device in groups[container_root] | arista.avd.natural_sort %}
 {%     if device | arista.avd.is_in_filter(local_var.device_filter) %}
+{%         set dev_cntr.value = dev_cntr.value + 1 %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2]: I think it's better to move this line of code to under below if condition
```{%         if hostvars[device]['is_deployed'] is arista.avd.defined(true) or hostvars[device]['is_deployed'] is not arista.avd.defined %}```

If this condition fails then also CVP_DEVICES will be returned None.

Comment (noredistribution) [reply ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2]: yea, that's how I did it initially, not sure why I changed my mind, but that's a good point, let me move it there, thanks!

File: ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2 [hunk 3]
```diff
@@ -5,15 +5,18 @@
...
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2]: I guess you forgot to delete this line.

File: ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2 [hunk 4]
```diff
@@ -5,15 +5,17 @@
 {% else %}
 {%    set local_var.device_filter = device_filter %}
 {% endif %}
+{% set dev_cntr = namespace(value=0) %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2:8]: Space required around ```=```. value = 0

File: ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2 [hunk 5]
```diff
@@ -5,15 +5,17 @@
 {% else %}
 {%    set local_var.device_filter = device_filter %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2]: missing indentation (1 space). Same above.

File: ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2 (hunk 4 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2:8]: No, according to PEP-8 this is correct. Adding spaces would be wrong.

**Review by ClausHolbechArista — DISMISSED** [review]
LGTM

**Review by tgodaA — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]
LGTM - we don't need an error message, since it is not necessarily an error that no devices where ready to be deployed. So we will just parse the empty list along to the cvp modules and they will return without any action being done.

**Pull request merged**